### PR TITLE
Refactory: Workflow Processor By Email based on new notification queue

### PR DESCRIPTION
**Pull request by yamelsenih**
Remove old implementation of WorkflowProcessor Class
Add new queue notification

This apply for:
- `org.compiere.server.WorkflowProcessor`

Related to: #3552

### CODE REVIEW

File: serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java [hunk 1]
```diff
@@ -374,43 +375,55 @@ private int sendEmail (MWFActivity activity, String AD_Message,
...
+			&& !list.contains(new Integer(m_model.getSupervisor_ID()))) {
+			counter++;
 			list.add (new Integer(m_model.getSupervisor_ID()));
 		}
-
+		Trx.run(transactionName -> {
+			//	Get instance for notifier
+			DefaultNotifier notifier = (DefaultNotifier) QueueLoader.getInstance().getQueueManager(DefaultNotifier.QUEUETYPE_DefaultNotifier)
+					.withContext(getCtx())
+					.withTransactionName(transactionName);
```
Comment (mckayERP) [inline serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java:412]: For clean code, extract this as a private method (something like getDefaultNotifierInstance) and get rid of the comment.

File: serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java [hunk 2]
```diff
@@ -374,43 +375,55 @@ private int sendEmail (MWFActivity activity, String AD_Message,
...
+			//	Send notification to queue
```
Comment (mckayERP) [inline serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java:413]: Unnecessary comment

File: serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java [hunk 3]
```diff
@@ -374,43 +375,55 @@ private int sendEmail (MWFActivity activity, String AD_Message,
...
+			notifier
+				.clearMessage()
+				.withApplicationType(DefaultNotifier.DefaultNotificationType_UserDefined)
+				.withText(subject)
+				.addAttachment(attachmentAsPDF.get())
+				.withDescription(message.get())
+				.withTableId(po.get_Table_ID())
+				.withRecordId(po.get_ID());
+			//	Add all recipients
```
Comment (mckayERP) [inline serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java:422]: Unnecessary comment

File: serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java [hunk 4]
```diff
@@ -374,43 +375,55 @@ private int sendEmail (MWFActivity activity, String AD_Message,
...
+			list.forEach(userId -> notifier.addRecipient(userId));
+			//	Add to queue
```
Comment (mckayERP) [inline serverRoot/src/main/server/org/compiere/server/WorkflowProcessor.java:424]: Unnecessary comment

**Review by mckayERP — COMMENTED** [review]
Sorry I don't have time to do more.  I like the clean code approach and my comments are related to that.  If you are refactoring a method, you can go even further and clean up the legacy code quite a bit.  This is just a suggestion.

Are there any tests for this new code?

**Review by rcapecce — COMMENTED** [review]
Hi @yamelsenih , can you add any step to reproduce this from a IDE and test this functionality?

**Comment by SusanneCalderon** [comment]
hi @yamelsenih 
I tested with Jetty, here is the log.
The Workflow Responsible is GardenUser, with an email configuration which works in other tests.

imagen

I don't know how to test it in eclipse, can you help me, please?

**Comment by yamelsenih** [comment]
Hi @SusanneCalderon this error is because the user that is allocated as supervisor not have email configured.

Please check it and also you can check the Notification Type for this user

**Pull request merged**